## Question 36 - Rolling to win!
* Question from Data Interview Qs https://www.interviewqs.com/ 

Suppose you are playing a game with a fair die (e.g. no bias for any one side), and your expected payout based on the # you roll is shown the table below. If you have to pay $1 to play, is it worth the cost to play?

In [8]:
import pandas as pd
Fairdie = pd.DataFrame({'Roll':[6,5,4,3,2,1],"Payout":[4,2,1,0,0,0]})
Fairdie = Fairdie[['Roll','Payout']]

In [9]:
Fairdie

Roll  Payout
0     6       4
1     5       2
2     4       1
3     3       0
4     2       0
5     1       0

each result of roll has probability of 1/6. Taking account the probability of each roll and payout, the expected payout should be calculated at below.

In [10]:
Fairdie['Probability']=1/6

In [11]:
Fairdie

Roll  Payout  Probability
0     6       4     0.166667
1     5       2     0.166667
2     4       1     0.166667
3     3       0     0.166667
4     2       0     0.166667
5     1       0     0.166667

In [12]:
Fairdie['ExpectedPayout']=Fairdie['Payout']*Fairdie['Probability']

In [13]:
Fairdie

Roll  Payout  Probability  ExpectedPayout
0     6       4     0.166667        0.666667
1     5       2     0.166667        0.333333
2     4       1     0.166667        0.166667
3     3       0     0.166667        0.000000
4     2       0     0.166667        0.000000
5     1       0     0.166667        0.000000

In [19]:
TotalExpectedPayout = Fairdie['ExpectedPayout'].sum(axis=0)

In [20]:
TotalExpectedPayout

1.1666666666666667

Each time the expected payout considering each numbers' probability is $1.167. Since it cost only $1 to play, we are expected to gain from playing.

## So play the game!